Snowflake.Data/Core/SFStatement.cs

```diff
@@ -3,20 +3,107 @@
  */
 
 using System;
-using System.Web;
-using Newtonsoft.Json.Linq;
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 using Snowflake.Data.Client;
-using Snowflake.Data.Core.FileTransfer;
 using Snowflake.Data.Log;
 using System.Threading;
 using System.Threading.Tasks;
 using System.Text;
+using System.ComponentModel;
 
 namespace Snowflake.Data.Core
 {
+    /// <summary>
+    /// The status types of the query.
+    /// </summary>
+    public enum QueryStatus
+    {
+        [StringAttr(value = "UNKNOWN")]

```

Comment:
let's not introduce "unknown" enum value just to throw an exception for case of an unknown value.
Someone could think that "unknown" is one of legal values.
Instead you can write it such a way:
```
            var statuses = Enum.GetValues(typeof(QueryStatus))
                .Cast<QueryStatus>()
                .Where(v => v.GetAttribute<StringAttr>().value.Equals(description, StringComparison.OrdinalIgnoreCase));
            return statuses.Any() ? statuses.First() : throw new Exception("The query status returned by the server is not recognized");
```

or maybe just something like that would work:
```
Enum.Parse(typeof(QueryStatus, description, true))
```

Reply:
Done, removed the `Unknown` enum and refactored using the above